sdk/core/core/inc/az_credentials.h

```diff
@@ -45,12 +46,21 @@ typedef struct
 {
   struct
   {
-    uint8_t token[_az_TOKEN_BUF_SIZE]; /*!< Base64-encoded token */
     int16_t token_length;
     int64_t expires_at_msec;
+    uint8_t token[_az_TOKEN_BUF_SIZE]; /*!< Base64-encoded token */
   } _internal;
 } _az_token;
 
+typedef struct
+{
+  struct
+  {
+    bool* volatile lock;
```

Comment:
Do we have any atomic assignment operations in C? Is using `volatile` sufficient to implement a locking mechanism that is thread-safe?

Reply:
Oh do I wish it had!!! :) https://www.drdobbs.com/parallel/volatile-vs-volatile/212701484

Reply:
(resolving)